### ISSUE TITLE
UserWarning:Detected call of `lr_scheduler.step()` before `optimizer.step()`.

### ISSUE THREAD

**Issue by navy63** [yuanjunchai/IKC#3]
19-10-08 11:06:23.167 - INFO: Start training from epoch: 0, iter: 0
C:\Users\mayn\Anaconda3\envs\pytorch\lib\site-packages\torch\optim\lr_scheduler.py:82: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule.See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
19-10-08 11:12:06.405 - INFO: <epoch:  0, iter:     100, lr:1.000e-04> l_pix: 1.0065e-01

**Comment by yuanjunchai** [comment]
Seemingly you are training SFTMD, have you changed train_SFTMD.yaml?

**Comment by navy63** [comment]
Yes, I am now training, single card 2080TI training, it seems that this takes 5 days, the training process can understand, do not understand is the test process, when testing, why we need to test the two process, and how the result is obtained.
example:
gpu_ids: [0]
datasets:
   train:
       dataroot_GT: E:/Dataset/Flickr2K/Flickr2K_train/HR/x4
  val:
      dataroot_GT: E:/Dataset/Set14_test/HR/x4

19-10-09 08:41:18.331 - INFO: <epoch:575, iter:  95,000> psnr: 26.535406
19-10-09 08:42:41.467 - INFO: <epoch:576, iter:  95,100, lr:1.355e-05> l_pix: 2.9932e-02
19-10-09 08:43:54.576 - INFO: <epoch:576, iter:  95,200, lr:1.347e-05> l_pix: 2.7815e-02
19-10-09 08:45:23.507 - INFO: <epoch:577, iter:  95,300, lr:1.338e-05> l_pix: 2.9500e-02
19-10-09 08:46:46.556 - INFO: <epoch:578, iter:  95,400, lr:1.330e-05> l_pix: 3.7441e-02

**Comment by yuanjunchai** [comment]
well, you could only test IKC or SFTMD(non-blind SR net), for more details please refer to the paper.
Thank you!

**Comment by navy63** [comment]
OK，Thank you!
